src/utils/errors/error.ts

```diff
@@ -0,0 +1,54 @@
+import { CustomErrorMapping, ErrorDefinition } from './types';
+import { STRING_PREFIX_AFTER_UNICODE_REPLACEMENT, RAILGUN_ERRORS, CUSTOM_ERRORS, INVALID_ASCII_REGEX } from './constants';
+import { isDefined } from '../util';
+
+const validAscii = (str: string) => {
+  return str.replace(
+    INVALID_ASCII_REGEX,
+    '',
+  );
+};
+
+const findMatchingError = (errorMessage: string, errorMapping: CustomErrorMapping): ErrorDefinition | null => {
+  const lowercaseMsg = errorMessage.toLowerCase();
+  
+  for (const [, errorDef] of Object.entries(errorMapping)) {
+    if (errorDef.matches.some(match => lowercaseMsg.includes(match))) {
+      return errorDef;
+    }
+  }
+  return null;
+}
+
+const isRailgunError = (cause: Error): boolean => cause.message.toLowerCase().includes('railgunsmartwallet')
+
+export const sanitizeError = (cause: Error): Error => {
+  if (isDefined(cause) && cause.message) {
```

Comment:
Personally, since the "else" here is shorter, early return would make the logic easier to read, eg.

```ts
if (cause?.message == null) return new Error('Unknown error. Please try again.', { cause });

if (isRailgunError(cause)) { /* ... */ }
```

Another question is, do we even need to check if `cause` is defined and has a `message` property? The type here says it must be an `Error`, so we perhaps only need to check the `message` being defined